src/N3DataFactory.js

```diff
@@ -256,6 +265,53 @@ export class Quad {
 }
 export { Quad as Triple };
 
+// QuadTerm constructor
+export class QuadTerm extends Term {
+  constructor(quad) {
+    super(`<<${quadToId(quad)}>>`);
+    this.quad = quad;
+  }
+
+  // ### The term type of this term
+  get termType() {
+    return 'QuadTerm';
+  }
+
+  // ### The text value of this literal
+  get value() {
+    return this.quad;
+  }
+}
+
+// ### Creates a string from a Quad
+function quadToId(quad) {
+  let result = [quad.subject, quad.predicate, quad.object].map(termToId).map(escape).join(' ');
```

Comment:
Let's avoid the creation of three arrays and be a bit more redundant with string interpolation, just for perf reasons.